smartcar/auth_client.py

```diff
@@ -28,23 +30,37 @@ def __init__(
         However, if neither an environment variable nor an argument is passed in, an exception will be raised.
 
         Args:
-            client_id (str): The application id, provided in the application
+            client_id (str, optional): The application id, provided in the application
                 dashboard
 
-            client_secret (str): The application secret, provided in the
+            client_secret (str, optional): The application secret, provided in the
                 application dashboard
 
-            redirect_uri (str): The URL to redirect to after the user accepts
+            redirect_uri (str, optional): The URL to redirect to after the user accepts
                 or declines the application's permissions. This URL must also be
                 present in the Redirect URIs field in the application dashboard
 
             test_mode (bool, optional): Launch the Smartcar auth flow in test mode. Defaults to false.
+                test_mode is now deprecated. Use mode instead
```

Comment:
```suggestion
            test_mode (bool, optional): Deprecated, please use `mode` instead. Launch the Smartcar auth flow in test mode.
```

Nit: better to state that it is deprecated earlier rather than later